File: app/Models/PricingPlan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

/**
 * @property int $id
 * @property string $name
 * @property string $slug
 * @property string|null $description
 * @property string $price
 * @property int|null $duration
 * @property string|null $billing_cycle
 * @property array|null $features
 * @property int $sort_order
 * @property bool $is_active
 * @property \Carbon\Carbon $created_at
 * @property \Carbon\Carbon $updated_at
 */
class PricingPlan extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'slug',
        'description',
        'price',
        'duration',
        'billing_cycle',
        'features',
        'sort_order',
        'is_active',
    ];

    protected $casts = [
        'price' => 'decimal:2',
        'features' => 'array',
        'is_active' => 'boolean',
    ];

    public function products(): BelongsToMany
    {
        return $this->belongsToMany(Product::class, 'pricing_plan_products')
            ->withPivot('quantity')
            ->withTimestamps();
    }
}

File: mobile-app/lib/src/models/pricing_plan.dart
import 'product.dart';

class PricingPlan {
  final int id;
  final String name;
  final String slug;
  final String? description;
  final double price;
  final String? duration;
  final String? billingCycle;
  final List<String> features;
  final bool isActive;
  final bool isPopular;
  final List<Product> products;

  const PricingPlan({
    required this.id,
    required this.name,
    required this.slug,
    this.description,
    required this.price,
    this.duration,
    this.billingCycle,
    this.features = const [],
    this.isActive = true,
    this.isPopular = false,
    this.products = const [],
  });

  factory PricingPlan.fromJson(Map<String, dynamic> json) {
    return PricingPlan(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      slug: json['slug'] as String? ?? '',
      description: json['description'] as String?,
      price: _d(json['price']),
      duration: json['duration'] as String?,
      billingCycle: json['billing_cycle'] as String?,
      features: (json['features'] as List<dynamic>?)
              ?.map((e) => e.toString())
              .toList() ??
          const [],
      isActive: json['is_active'] as bool? ?? true,
      isPopular: json['is_popular'] as bool? ?? false,
      products: (json['products'] as List<dynamic>?)
              ?.map((e) => Product.fromJson(e as Map<String, dynamic>))
              .toList() ??
          const [],
    );
  }

  static double _d(dynamic v) {
    if (v is double) return v;
    if (v is int) return v.toDouble();
    if (v is String) return double.tryParse(v) ?? 0;
    return 0;
  }
}
